gap/examples.gi

```diff
@@ -370,3 +370,51 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallMethod(KnightsGraphCons,
+"for IsMutableDigraph and two positive integers",
+[IsMutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D, moveOffSets, coordinates, targetCoordinate, p, i, j;
+  D := EmptyDigraph(IsMutableDigraph, n * k);
+  moveOffSets := [[2, 1], [-2, 1], [2, -1], [-2, -1],
+  [1, 2], [-1, 2], [1, -2], [-1, -2]];
+  coordinates := [];
+  p := 0;
+
+  for i in [1 .. k] do
+    for j in [1 .. n] do
+      p := p + 1;
+      coordinates[p] := [i, j];
+    od;
+  od;
+
+  for i in coordinates do
+    for j in moveOffSets do
+      targetCoordinate := [i[1] + j[1], i[2] + j[2]];
+      if targetCoordinate in coordinates then
+        DigraphAddEdge(D, [Position(coordinates, i),
+        Position(coordinates, targetCoordinate)]);
+      fi;
+    od;
+  od;
+  return D;
+end);
+
+InstallMethod(KnightsGraphCons,
+"for IsImmutableDigraph and two positive integers",
+[IsImmutableDigraph, IsPosInt, IsPosInt],
+function(filt, n, k)
+  local D;
+  D := MakeImmutable(KnightsGraphCons(IsMutableDigraph, n, k));
+  SetIsMultiDigraph(D, false);
+  SetIsSymmetricDigraph(D, true);
```

Comment:
Could you work out precisely when `D` is connected? It's probably something like
```suggestion
  SetIsSymmetricDigraph(D, true);
  SetIsConnectedDigraph(D, n > 2 and m > 2 and not (m = 3 and n = 3));
```
That would be nice to add.